src/en/ScribbleHub.lua

```diff
@@ -124,10 +124,12 @@ return {
 		local title = chap:selectFirst(".chapter-title"):text()
 		chap = chap:getElementById("chp_raw")
 
-		-- remove empty <p> tags
+		-- Remove <p></p>.
 		local toRemove = {}
 		chap:traverse(NodeVisitor(function(v)
-			if v:tagName() == "p" and v:text() == "" then
+			if v:tagName() == "p" and v:childNodeSize() == 0 then -- and v:text() == "" then
+				-- childNodeSize() == 0 iff <tag></tag>, therefore not triggered by <p> </p>.
+				-- Avoids removal of things like <p><img src="some link"></p> though.
```

Comment:
```suggestion
			if v:tagName() == "p" and v:childrenSize() == 0 and v:text() == "" then
```
Just checking that there are no *Element* child nodes (`childrenSize() == 0`) should be fine.